src/client/theme-default/components/NavBarLinks.ts

```diff
@@ -40,14 +41,49 @@ export default {
       }
       return null
     })
+
+    const localeCandidates = computed(() => {
+      const locales = siteData.value.themeConfig.locales
+      const localeKeys = Object.keys(locales)
```

Comment:
At here, we must check whether if the `locales` exists or not.

```js
const locales = siteData.value.themeConfig.locales
if (!locales) {
  return null
}
const localeKeys = Object.keys(locales)
```

If the config doesn't contain any locales, `Object.keys(locales)` will fail due to `locales` is `undefined` in that case.